README.md

```diff
@@ -7,11 +7,60 @@
+
+## Unmounting
+At the unmounting stage, the component gets deleted and cleared out of the page. The only lifecycle hook at this stage is **componentWillUnmount**, which is called just before the component gets deleted. This is used to clear out any stuff set up in **componentDidMount**. For example, if you had a component that displays the weather data in your home town, you might have set it up to re-fetch the updated weather information every 10 seconds in **componentDidMount**. When the component gets deleted, you wouldn't want to continue doing this data-fetching, so you'd have to get rid of what was set up in **componentWillUnmount**. We will look at a detailed example of how it's done in the next module.
+
+## Updating
+Whenever a component's state or props are changed, it gets re-rendered on the page. That's the beauty of React components - they're quick to *react* to changes. A re-render could be triggered when a user interacts with the component, or if new data (props or state) are passed in. For example, going back to the chat window on Facebook example, whenever you press "send" on a message, the ChatWindow component gets re-rendered as it needs to display an extra message. Whenever a re-render is triggered, there is a whole host of lifecycle hooks which get called. You can choose to use any of these to decide how your React component should respond to changes.
+
+**componentWillReceiveProps** is invoked when the props the parent is passing into the component have changed. You could use this to change the component's state based on the new props.
+
+**shouldComponentUpdate** is invoked just before the component is about to re-render. At this stage, you can compare the old and new props and state and prevent unnecessary re-renders: if the change in state or props don't actually alter the component that's being shown to the user, there is no point "repainting" it as it is an unnecessary drain on your website.
+
+**componentWillUpdate** is called just after **shouldComponentUpdate** has finished and just before the new component gets rendered. You would usually use this method to set up integrations with third party libraries.
+
+**componentDidUpdate** is called just after the new component had been rendered. You will have access to the previous props and state as well as the current ones, and you can use this method to update any third party libraries if they happen to need an update due to the re-render.
+
+## Summary
+If you need a lifecycle hook, there's one for your every need! Don't worry if you couldn't quite take it all in: we will go through all of these in detail in the next few modules.
```

Comment:
> Don't worry if you couldn't quite take it all in: we will go through all of these in detail in the next few modules.

See lesson decoupling note above, we generally try to avoid this